pkg/kv/kvserver/store_remove_replica.go

```diff
@@ -88,64 +88,67 @@ func (s *Store) removeInitializedReplicaRaftMuLocked(
 
 	}
 
-	// Sanity checks before committing to the removal by setting the
-	// destroy status.
-	var desc *roachpb.RangeDescriptor
+	// Another sanity check that the Replica that we're planning to remove is
+	// still the one in the Store.
 	{
+		existing, err := s.GetReplica(rep.RangeID)
```

Comment:
This has to be ordered after the `return nil, nil` in the closure below. Thanks tests for pointing this out.

Reply:
Actually I like this new code more, we seem to be removing already removed replicas - the old conditional was clearly put in to make that allowed, but it seems worth understanding how we end up in situations in which we're replicaGC'ing a previously replicaGC'ed replica. Perhaps some race in the replicate queue?